src/frontend/components/Felleskomponenter/MånedÅrVelger/MånedÅrVelger.tsx

```diff
@@ -0,0 +1,121 @@
+
+import { useApp } from '../../../context/AppContext';
+import { useSpråk } from '../../../context/SpråkContext';
+import { ISODateString } from '../../../typer/common';
+import { ESanitySteg } from '../../../typer/sanity/sanity';
+
+interface IProps {
+    tidligsteValgbareMåned?: Date;
+    senesteValgbareMåned?: Date;
+    label: React.ReactNode;
+    felt: Felt<ISODateString>;
+    visFeilmeldinger?: boolean;
+    disabled?: boolean;
+    dagIMåneden: DagIMåneden;
+    kanIkkeVæreFortid?: boolean;
+    kanIkkeVæreFremtid?: boolean;
+}
+
+/* TODO
+- vurder å lagre som Date og gjøre parsing/format mot mellomlagring og innsending - i neste runde
+*/
+
+export enum Feilmelding {
+    FØR_MIN_DATO = 'FØR_MIN_DATO',
+    ETTER_MAKS_DATO = 'ETTER_MAKS_DATO',
+    UGYLDIG_DATO = 'UGYLDIG_DATO',
+    DATO_IKKE_I_FORTID = 'DATO_IKKE_I_FORTID',
+    DATO_IKKE_I_FREMTID = 'DATO_IKKE_I_FREMTID',
+}
+
+export enum DagIMåneden {
+    FØRSTE_DAG = 'FØRSTE_DAG',
+    SISTE_DAG = 'SISTE_DAG',
+}
+
+export const MånedÅrVelger: React.FC<IProps> = ({
+    tidligsteValgbareMåned,
+    senesteValgbareMåned,
+    label,
+    felt,
+    visFeilmeldinger = false,
+    disabled = false,
+    dagIMåneden = DagIMåneden.FØRSTE_DAG,
+    kanIkkeVæreFortid,
+    kanIkkeVæreFremtid,
+}) => {
+    const { valgtLocale } = useSpråk();
+    const { tekster, plainTekst } = useApp();
+    const [error, setError] = useState<Feilmelding | undefined>(undefined);
+    const formateringsfeilmeldinger = tekster()[ESanitySteg.FELLES].formateringsfeilmeldinger;
+
+    const nullstillOgSettFeilmelding = (feilmelding: Feilmelding) => {
+        if (error !== feilmelding) {
+            setError(feilmelding);
+            felt.nullstill();
+        }
+    };
+
+    const feilmeldinger: Record<Feilmelding, string> = {
+        UGYLDIG_DATO: plainTekst(formateringsfeilmeldinger.ugyldigManed),
+        FØR_MIN_DATO: plainTekst(formateringsfeilmeldinger.datoErForForsteGyldigeTidspunkt),
+        ETTER_MAKS_DATO: plainTekst(formateringsfeilmeldinger.datoErEtterSisteGyldigeTidspunkt),
+        DATO_IKKE_I_FORTID: plainTekst(formateringsfeilmeldinger.datoKanIkkeVareIFortid),
+        DATO_IKKE_I_FREMTID: plainTekst(formateringsfeilmeldinger.datoKanIkkeVareIFremtid),
+    };
+
+    const { monthpickerProps, inputProps } = useMonthpicker({
+        fromDate: tidligsteValgbareMåned,
+        toDate: senesteValgbareMåned,
+        locale: valgtLocale,
+        onMonthChange: (dato: Date | undefined): void => {
+            if (dato === undefined) {
+                felt.nullstill();
+            } else {
+                if (dagIMåneden === DagIMåneden.FØRSTE_DAG) {
+                    felt.validerOgSettFelt(formatISO(dato, { representation: 'date' }));
+                } else {
+                    felt.validerOgSettFelt(
+                        formatISO(lastDayOfMonth(dato), { representation: 'date' })
+                    );
+                }
+            }
```

Comment:
```suggestion
            if (dato === undefined) {
                felt.nullstill();
            } else {
                felt.validerOfSettFelt(formatISO(formatterTilRiktigDagIMåneden(dato)))
            }
```


Må da legge til denne funksjonen et annet sted i samme fil
```
const formatterTilRiktigDagIMåneden = (dato: Date): Date => {
        if (dagIMåneden === DagIMåneden.FØRSTE_DAG) return startOfMonth(dato);
        else if (dagIMåneden === DagIMåneden.SISTE_DAG) return endOfMonth(dato);
        else return dato;
    };

```